src/helpers/gateway/mod.rs

```diff
@@ -2,30 +2,42 @@ mod receive;
 mod send;
 mod transport;
 
-pub use receive::ReceivingEnd;
 pub use send::SendingEnd;
 
 use std::{fmt::Debug, num::NonZeroUsize};
 
 use crate::{
     helpers::{
         gateway::{
-            receive::GatewayReceivers, send::GatewaySenders, transport::RoleResolvingTransport,
+            receive::{GatewayReceivers, ReceivingEnd as ReceivingEndBase},
+            send::GatewaySenders,
+            transport::RoleResolvingTransport,
         },
-        transport::TransportImpl,
-        ChannelId, Message, Role, RoleAssignment, TotalRecords,
+        ChannelId, Message, Role, RoleAssignment, TotalRecords, Transport,
     },
     protocol::QueryId,
 };
 #[cfg(all(feature = "shuttle", test))]
 use shuttle::future as tokio;
 
-/// Gateway into IPA Infrastructure systems. This object allows sending and receiving messages
-pub struct Gateway {
+#[cfg(any(feature = "test-fixture", test))]
+pub type TransportImpl = crate::test_fixture::network::InMemoryTransport;
+#[cfg(not(any(feature = "test-fixture", test)))]
+pub type TransportImpl = crate::helpers::transport::DummyTransport;
+
+pub type Gateway = GatewayBase<TransportImpl>;
+pub type ReceivingEnd<M> = ReceivingEndBase<TransportImpl, M>;
+
+/// Gateway into IPA Infrastructure systems. This object allows sending and receiving messages.
+/// As it is generic over network/transport layer implementation, type alias [`Gateway`] should be
+/// used to avoid carrying `T` over.
+///
+/// [`Gateway`]: crate::helpers::Gateway
+pub struct GatewayBase<T: Transport> {
```

Comment:
`pub struct Gateway<T: Transport = TransportImpl>` is also a possibility.

(I assume you're using the type parameter to catch accidental use of non-generic methods on whichever impl is configured in a particular build? If the concern is just making sure that `InMemoryTransport` and `HttpTransport` share a common interface, there's an idiom I can suggest to enforce that without having to code against the `Transport` trait, but it won't prevent non-trait functionality from leaking through.)